src/main/java/de/tum/in/www1/artemis/web/rest/dto/SubmissionDTO.java

```diff
@@ -0,0 +1,35 @@
+package de.tum.in.www1.artemis.web.rest.dto;
+
+import java.time.ZonedDateTime;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.ProgrammingSubmission;
+import de.tum.in.www1.artemis.domain.Submission;
+import de.tum.in.www1.artemis.domain.enumeration.SubmissionType;
+
+/**
+ * DTO containing {@link Submission} information.
+ * This does not include large reference attributes in order to send minimal data to the client.
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record SubmissionDTO(Long id, Boolean submitted, SubmissionType type, Boolean exampleSubmission, ZonedDateTime submissionDate, String commitHash, Boolean buildFailed,
+        Boolean buildArtifact, DomainObjectIdDTO participation, String submissionExerciseType) {
+
+    /**
+     * Converts a Submission into a SubmissionDTO.
+     *
+     * @param submission to convert
+     * @return the converted DTO
+     */
+    public static SubmissionDTO of(Submission submission) {
+        if (submission instanceof ProgrammingSubmission programmingSubmission) {
```

Comment:
I think some explanation on why the distinction is necessary would be more valuable than the interface comment here.